torch/nn/parallel/distributed.py

```diff
@@ -575,6 +576,15 @@ def no_sync(self):
             self.require_backward_grad_sync = old_require_backward_grad_sync
 
     def forward(self, *inputs, **kwargs):
+        # Calling _rebuild_buckets before forward compuation,
+        # It may allocate new buckets before deallocating old buckets
```

Comment:
> It may allocate new buckets before deallocating old buckets

Why? at this point, looks like old buckets are no longer necessary. Is it possible to first delete old buckets and then allocate new buckets?

Reply:
grad points to old buckets, we can not simply delete old buckets as users may not want to delete the grads